packages/paste-core/components/sidebar/src/navigation/SidebarNavigationDisclosureHeading.tsx

```diff
@@ -22,10 +23,26 @@ export interface SidebarNavigationDisclosureHeadingProps extends React.Component
 
 const StyledDisclosureHeading = React.forwardRef<HTMLDivElement, SidebarNavigationDisclosureHeadingProps>(
   ({children, element = 'SIDEBAR_NAVIGATION_DISCLOSURE_HEADING', selected, icon, ...props}, ref) => {
+    const {collapsed, variant} = React.useContext(SidebarContext);
     const [shouldIconMove, setShouldIconMove] = React.useState(false);
     const {nested} = React.useContext(SidebarNavigationDisclosureContext);
     const isExpanded = props['aria-expanded'];
     const theme = useTheme();
+    const isCompact = variant === 'compact';
+    const [visible, setVisible] = React.useState(!isCompact ? true : !isExpanded);
```

Comment:
Note: I need a variable that copies `collapsed` but is set after a delay. This hook and timeout is how I achieve this. I use this to adjust whiteSpace and opacity AFTER the sidebar is expanded, so that we don't see word-wrapping occur throughout the expand process. It only triggers at the end of the animation once. This makes the animation end look like a single clean-up step rather than the entire animation to cause several height reflows.